src/Core/Grid/Query/ProductCombinationQueryBuilder.php

```diff
@@ -137,21 +137,36 @@ private function getCombinationsQueryBuilder(ProductCombinationFilters $productC
 
         if (null === $productCombinationFilters->getOrderBy()) {
             $qb->addOrderBy('id_product_attribute', 'asc');
+        } elseif ('stock_quantity' === $productCombinationFilters->getOrderBy()) {
+            $qb
+                ->addSelect('pa.quantity AS stock_quantity')
+                ->leftJoin(
+                    'pa',
+                    $this->dbPrefix . 'stock_available',
+                    'sa',
+                    'pa.id_product_attribute = sa.id_product_attribute'
+                )
+            ;
         }
 
         return $qb;
     }
 
     /**
      * @param int $productId
-     * @param int[] $attributeIds
+     * @param array<int, array<int>> $attributeGroups
      *
      * @return int[]
      */
-    private function getCombinationIdsByAttributeIds(int $productId, array $attributeIds): array
+    private function getCombinationIdsByAttributeIds(int $productId, array $attributeGroups): array
```

Comment:
@PrestaShop/prestashop-core-developers I you guys have a better way to handle this query, I was not able to do it with pure SQL/DQL

So I had to fetch all the matching combinations by attributes, and then manage the filtering with PHP, the rule here is:
- an OR among the list of attributes
- but and AND for all selected groups

So basically it must match at least one of the elements in each attribute groups 🤔

Reply:
I think it's quite ok to filter in php